File: app/schemas/media.py
import uuid
from datetime import datetime

from pydantic import BaseModel


class MediaResponse(BaseModel):
    id: uuid.UUID
    tree_id: uuid.UUID
    story_id: uuid.UUID | None
    person_id: uuid.UUID | None
    uploaded_by_id: uuid.UUID
    filename: str
    original_filename: str
    mime_type: str
    size_bytes: int | None
    media_type: str
    caption: str | None
    created_at: datetime

    model_config = {"from_attributes": True}

File: frontend/src/types/media.ts
export interface Media {
  id: string;
  tree_id: string;
  story_id: string | null;
  person_id: string | null;
  uploaded_by_id: string;
  filename: string;
  original_filename: string;
  mime_type: string;
  size_bytes: number | null;
  media_type: string;
  caption: string | null;
  created_at: string;
}
